.github/workflows/rust.yml

```diff
@@ -70,6 +70,10 @@ jobs:
         run: |
           rustup toolchain install stable --no-self-update
           rustup default stable
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v3
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
...Why does installing protoc need the repo token?

Reply:
Hmm, supposedly to query Github for the latest version...

Is there some other source for protoc that doesn't rely on a third party action?

Reply:
- We can just do curl+unzip but it will require some logic to choose the right binary in order to work cross-platform.
- Another option is to install `grpcio-tools` from pypi and then alias `protoc` to `python -m grpc_tools.protoc`, but that obviously requires python to be installed.

Reply:
I suppose system protoc would be too old? (And Conda too heavy?)

Reply:
pixi should be fine, I think. I can add pixi.toml in rust directory, we can even install rust from there instead of using rustup iirc. Not sure if just protoc justifies pixi adoption, but it would be the simplest alternative.

Reply:
system protoc might not be that old, but it will a different version in a different os.

Reply:
I suppose, (1) what actually needs protoc here? (2) if the version matters, we could check generated code in instead?

Reply:
(1) substrait crate depends on protoc
(2) I guess, but if we still need to write some logic to install the system protoc (check whether to use apt or brew dependending on the os), we might as well download the binaries instead.

Reply:
Looks like this is a no-go :laughing:. judging from the action code, even if I pin an exact version, the actions still tries to fetch all versions, presumably to validate.. seems like there's no way to make it work w/o a token.

Reply:
@lidavidm Finally got it working with pre-compiled binaries